### PR TITLE
Yield events from the feed individually

### PR DESCRIPTION
**Pull request by TyMarc**

### CODE REVIEW

File: packages/flare/bin/flare.py [hunk 1]
```diff
@@ -99,16 +96,11 @@ def _retrieve_event_feed_metadata(
...
-                url=f"/firework/v2/activities/{event['metadata']['uid']}"
-            )
-            enriched_event = event_response.json()
-            self.logger.debug(enriched_event)
-            return enriched_event
-        else:
-            return event
+    def _retrieve_full_event_from_uid(self, *, uid: str) -> dict:
+        event_response = self.flare_client.get(url=f"/firework/v2/activities/{uid}")
+        enriched_event = event_response.json()
```
Comment (markkasaboski) [inline packages/flare/bin/flare.py]: Probably change `enriched_event` with just `event`

**Review by markkasaboski — APPROVED** [review]
Just one nit about variable name. LGTM

File: packages/flare/bin/flare.py [hunk 2]
```diff
@@ -99,16 +96,11 @@ def _retrieve_event_feed_metadata(
...
-            )
-            enriched_event = event_response.json()
-            self.logger.debug(enriched_event)
-            return enriched_event
-        else:
-            return event
+    def _retrieve_full_event_from_uid(self, *, uid: str) -> dict:
+        event_response = self.flare_client.get(url=f"/firework/v2/activities/{uid}")
+        event = event_response.json()
+        self.logger.debug(event)
```
Comment (aviau) [inline packages/flare/bin/flare.py:98]: These are very big logs. They are not included in prod right?

File: packages/flare/bin/flare.py [hunk 3]
```diff
@@ -44,34 +44,31 @@ def __init__(self, *, api_key: str, tenant_id: Optional[int] = None) -> None:
...
-                    try:
-                        full_event_feed["items"].append(
-                            self._enrich_event_from_metadata_uid(event=event)
-                        )
-                    except Exception as e:
-                        self.logger.error(f"Exception={e}")
-                        full_event_feed["items"].append(event)
-                    # Rate limiting.
+            next_token = event_feed["next"]
+            for event in event_feed["items"]:
```
Comment (aviau) [inline packages/flare/bin/flare.py:61]: you can probably simplify a little bit:
```python
if ingest_metadata_only:
    event = self._retrieve_full_event_from_uid(uid=event["metadata"]["uid"])
yield event, next_token
```

File: packages/flare/bin/flare.py [hunk 4]
```diff
@@ -99,16 +96,11 @@ def _retrieve_event_feed_metadata(
...
-                url=f"/firework/v2/activities/{event['metadata']['uid']}"
-            )
-            enriched_event = event_response.json()
-            self.logger.debug(enriched_event)
-            return enriched_event
-        else:
-            return event
+    def _retrieve_full_event_from_uid(self, *, uid: str) -> dict:
+        event_response = self.flare_client.get(url=f"/firework/v2/activities/{uid}")
+        event = event_response.json()
```
Comment (aviau) [inline packages/flare/bin/flare.py]: Are you sure we shouldn't return event_response["event"] here? (or whatever the envelope is)

File: packages/flare/tests/bin/test_ingest_events.py [hunk 5]
```diff
@@ -209,29 +209,32 @@ def test_fetch_feed_expect_feed_response(
...
+    }
+    expected_items = [first_item, second_item]
     flare_api_mock_instance = flare_api_mock.return_value
-    flare_api_mock_instance.retrieve_feed.return_value = iter([expected_return_value])
+    flare_api_mock_instance.retrieve_feed_events.return_value = iter(
+        [(first_item, next), (second_item, next)]
+    )
 
-    for feed_event in fetch_feed(
+    events = []
```
Comment (aviau) [inline packages/flare/tests/bin/test_ingest_events.py:225]: untyped list.

File: packages/flare/bin/flare.py (hunk 2 above)
Comment (TyMarc) [reply packages/flare/bin/flare.py:98]: they are not, neither locally unless you change your log level to debug. The minimum is Info

File: packages/flare/bin/flare.py (hunk 4 above)
Comment (TyMarc) [reply packages/flare/bin/flare.py]: Do you mean event_response["activity"]?

Comment (aviau) [reply packages/flare/bin/flare.py]: Yes

**Pull request merged**